packages/web/src/common/store/notifications/parseAndProcessNotifications.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  ID,
+  waitForRead,
+  Notification,
+  NotificationType,
+  Entity,
+  reactionsUIActions,
+  waitForAccount,
+  Track,
+  Achievement,
+  Remix,
+  accountSelectors
+} from '@audius/common'
+import moment from 'moment'
+import { all, call, put, select } from 'typed-redux-saga'
+
+import { retrieveCollections } from '../cache/collections/utils'
+import { retrieveTracks } from '../cache/tracks/utils'
+import { fetchUsers } from '../cache/users/sagas'
+
+const { fetchReactionValues } = reactionsUIActions
+const { getUserId } = accountSelectors
+
+// The initial user count to load in for each notification
+// NOTE: the rest are loading in in the user list modal
+export const USER_INITIAL_LOAD_COUNT = 9
+
+const getTimeAgo = (now: moment.Moment, date: string) => {
+  const notifDate = moment(date)
+  const weeksAgo = now.diff(notifDate, 'weeks')
+  if (weeksAgo) return `${weeksAgo} Week${weeksAgo > 1 ? 's' : ''} ago`
+  const daysAgo = now.diff(notifDate, 'days')
+  if (daysAgo) return `${daysAgo} Day${daysAgo > 1 ? 's' : ''} ago`
+  const hoursAgo = now.diff(notifDate, 'hours')
+  if (hoursAgo) return `${hoursAgo} Hour${hoursAgo > 1 ? 's' : ''} ago`
+  const minutesAgo = now.diff(notifDate, 'minutes')
+  if (minutesAgo) return `${minutesAgo} Minute${minutesAgo > 1 ? 's' : ''} ago`
+  return 'A few moments ago'
```

Comment:
I thought moment had some stuff built in for this. Do we just not quite agree with the language and categories?

Reply:
yeah i bet there is, this was all a copy from existing code... @jowlee thoughts?